File: ResourceManager/Services/Abstractions/IDataServiceManager.cs
namespace ResourceManager.Services.Abstractions
{
    public interface IDataServiceManager
    {
        IUserDataService UserDataService { get; }
        IRoleDataService RoleDataService { get; }
        IPermissionDataService PermissionDataService { get; }
        IPasswordDataService PasswordDataService { get; }
        IBookDataService BookDataService { get; }
        IRepositoryDataService RepositoryDataService { get; }
        IArticleDataService ArticleDataService { get; }
        IActivityDataService ActivityDataService { get; }
        ITagDataService TagDataService { get; }
        INoteDataService NoteDataService { get; }
    }
}

File: ResourceManager/Services/Abstractions/IDataServiceManager.vb
Namespace ResourceManager.Services.Abstractions
    Public Interface IDataServiceManager
        ReadOnly Property UserDataService As IUserDataService
        ReadOnly Property RoleDataService As IRoleDataService
        ReadOnly Property PermissionDataService As IPermissionDataService
        ReadOnly Property PasswordDataService As IPasswordDataService
        ReadOnly Property BookDataService As IBookDataService
        ReadOnly Property RepositoryDataService As IRepositoryDataService
        ReadOnly Property ArticleDataService As IArticleDataService
        ReadOnly Property ActivityDataService As IActivityDataService
        ReadOnly Property TagDataService As ITagDataService
        ReadOnly Property NoteDataService As INoteDataService
    End Interface
End Namespace
